src/units/gender.hpp

```diff
@@ -16,11 +16,42 @@
 
 #include <iosfwd>
 
-enum class unit_gender {
-	MALE, FEMALE, NUM_GENDERS
+class t_string;
+class unit_gender;
+
+class unit_gender {
+public:
+	static const unit_gender& male();
+	static const unit_gender& female();
+	static constexpr int num_genders(){
+		return 2;
+	}
+
+	static const unit_gender* from_string(const std::string&);
+	static const unit_gender& from_string(const std::string&, const unit_gender& fallback);
+	static const unit_gender* from_int(int);
```

Comment:
I don't think I really like the idea of passing around pointers to genders; I'd rather it be a lightweight type that can be cheaply copied.

Reply:
With pointers, it's just as cheap as the original version, and lacks the special `NUM_GENDERS` value.

Other options would be `MAKE_ENUM` or just the first commit of this branch.